src/test/java/uk/gov/companieshouse/chipsrestinterfacesconsumer/consumer/IncomingMessageConsumerUnitTest.java

```diff
@@ -59,23 +75,31 @@ void testReadValidMessage() {
         message.setValue("{}".getBytes());
         messages.add(message);
         when(consumer.consume()).thenReturn(messages);
+
         Collection<ChipsKafkaMessage> result = incomingMessageConsumer.read();
+
         assertEquals(1, result.size());
         assertNotNull(result.iterator().next());
     }
 
     @Test
     void testDeserializeExceptionIsCaught() {
+        String messageValue = "bad message";
         List<Message> messages = new ArrayList<>();
         Message message = new Message();
-        message.setValue("bad message".getBytes());
+        message.setValue(messageValue.getBytes());
         messages.add(message);
         when(consumer.consume()).thenReturn(messages);
 
         incomingMessageConsumer.read();
-        assertThatThrownBy(() -> incomingMessageConsumer.deserialise(message))
-                .isInstanceOf(IOException.class);
-        // TODO mock logger (autowire it in class being tested) and verify it is called when exception caught in read()
+
+        verify(logger, times(1)).info(anyString());
+        verify(logger, times(1)).error(
```

Comment:
is verifying the logger is called really worth adding to the unit test?

Reply:
i'd say yes, if an error is thrown then it has to be logged. By adding this test we make sure it does get logged. I had to add the info check before as error isn't the first method to be called on the logger mock, info was, so had to verify that before the error call.